### PR TITLE
TDEVOPS-827 | Master CI

### PR DESCRIPTION
**Pull request by varenyatessell**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

Dependent Services #
- [ ] Governance
- [ ] IAM
- [ ] Deployer
- [ ] DB Systems
- [ ] Tenant
- [ ] Software Library
- [ ] Backup
- [ ] Conductor
- [ ] Monitoring
- [ ] DMM

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Environment**:

- [ ] Dev
- [ ] Staging

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/master_read_codeowners.yml
Comment (sundeeptessell) [inline .github/workflows/master_read_codeowners.yml:69]: Shouldn't this condition be outside of inner loop?

File: .github/workflows/master_read_codeowners.yml
Comment (sundeeptessell) [inline .github/workflows/master_read_codeowners.yml:67]: Once you have a match, `break`

File: .github/workflows/master_read_codeowners.yml
Comment (sundeeptessell) [reply .github/workflows/master_read_codeowners.yml:69]: With `break` and `for-else` :

```
for file_path in changed_files_list:
    for pattern, team_name in team_patterns.items():
        if (file_path.startswith(pattern) or file_path.startswith(pattern[:-2])) and pattern != '*':
            changed_teams.update(team_name)
            break
    else
        changed_teams.update(team_patterns['*'])
```

File: .github/workflows/master_check_approval.yml
Comment (sundeeptessell) [inline .github/workflows/master_check_approval.yml:36]: Remember to undo the `varenyatessell` part before merge

File: .github/workflows/master_check_label.yml
Comment (sundeeptessell) [inline .github/workflows/master_check_label.yml:19]: You are assuming that it will be teams only ; But if the `CODEOWNERS` file has username this will fail

File: .github/workflows/master_approve.yml
Comment (sundeeptessell) [inline .github/workflows/master_approve.yml:28]: Help me understand this.

File: .github/workflows/master_check_label.yml
Comment (varenyatessell) [reply .github/workflows/master_check_label.yml:24]: We already discussed this in api-spec repo that having specific users won't be a good decision in case the user leaves, so better to have a separate team for that user

File: .github/workflows/master_pr.yml
Comment (sundeeptessell) [inline .github/workflows/master_pr.yml:71]: Want to see how you validated this. Please share the link to the test done for this.

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:23]: When will this be the case? I thought you told all workflows are on `pull_request` event?

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:29]: Same here

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:34]: Let us define it at the top (i.e., always, without any condition).

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:44]: Can we put this in variable and use it - `^(TS-|SRE-|TDEVOPS-|TOPS-)` ; Then if a new project is added, we do not have to make Changes in code.

I added a variable `SUPPORTED_JIRA_PROJECTS` ; Please test it

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:45]: If the variable works. then lets just print the variable value in message instead of hard-coding the project names.

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:49]: Same here

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:215]: Why `env.NEXUS_URL` not used here?

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:270]: Why this condition?

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:274]: Line 274 - 290 are same as 237 to 251;  Can we make it as one step with `response` saved as output of that step and use it as needed.

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:300]: This loop can be merged into the loop of 291 - 298

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:331]: remove?

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:337]: Same here. This block 337 - 353 is repeated

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:365]: Move this step above, so all db-migration related steps are together.

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:381]: This code has a bug. How was this tested?

File: .github/workflows/master_dev_checks.yml
Comment (sundeeptessell) [inline .github/workflows/master_dev_checks.yml:363]: Set output variable name in plural as there are multiple files

File: .github/workflows/master_check_version.yml
Comment (sundeeptessell) [inline .github/workflows/master_check_version.yml:130]: I defined below variables. Let us use them
```
NEXUS_PUSH_REPOS_PY
NEXUS_PUSH_REPOS_M2
NEXUS_PUSH_REPOS_RAW
NEXUS_PUSH_REPOS_HELM
```

File: .github/workflows/master_auto_merge.yml
Comment (sundeeptessell) [inline .github/workflows/master_auto_merge.yml:13]: For checkout action I see the latest version is `v4.1.1` ; Can you check if we can use it without any other change/impact. If it fails then open a JIRA and we can upgrade later.

Comment (sundeeptessell) [reply .github/workflows/master_auto_merge.yml:13]: Ignore. Create a separate JIRA to upgrade all public actions we are using.

File: .github/workflows/master_build_merge.yml
Comment (sundeeptessell) [inline .github/workflows/master_build_merge.yml:28]: Repo name will be available by default under env variable `GITHUB_REPOSITORY`;

File: .github/workflows/master_build_merge.yml
Comment (sundeeptessell) [inline .github/workflows/master_build_merge.yml:102]: Variable `NEXUS_SERVER_ENDPOINT` defined with value `nexus.tessell.cloud`; You can use the same at all places.

File: .github/workflows/master_build_merge.yml
Comment (sundeeptessell) [inline .github/workflows/master_build_merge.yml:105]: Same here. Use `GITHUB_REPOSITORY` default env variable.

File: .github/workflows/master_build_merge.yml
Comment (sundeeptessell) [inline .github/workflows/master_build_merge.yml:104]: I defined variable `DOCKERHUB_ORG` ; Get the value from `vars.DOCKERHUB_ORG`

File: .github/workflows/master_dev_checks.yml
Comment (varenyatessell) [reply .github/workflows/master_dev_checks.yml:381]: Noticed and fixed the bug, this code I just copied from existing harness code written before me

File: .github/workflows/master_build_merge.yml
Comment (sundeeptessell) [inline .github/workflows/master_build_merge.yml:103]: Defined two variables:
```
NEXUS_REPO_TESSELLOPS_ARTIFACTS
NEXUS_REPO_TESSELL_ARTIFACTS
```

File: .github/workflows/master_build_pr.yml
Comment (sundeeptessell) [inline .github/workflows/master_build_pr.yml:81]: Making same mistake with `mvnw` command vs `mvnw` Function. Although its working for you now since you used `./mvnw`, for anyone else if they try without `./` it will result in recursion.  Please use a different name for function.

File: .github/workflows/master_build_pr.yml
Comment (sundeeptessell) [inline .github/workflows/master_build_pr.yml:88]: Why repeat the command. Call the function of mvnw

File: .github/workflows/master_auto_merge.yml
Comment (varenyatessell) [reply .github/workflows/master_auto_merge.yml:13]: https://tessell.atlassian.net/browse/TDEVOPS-1248

File: .github/workflows/master_build_merge.yml
Comment (varenyatessell) [reply .github/workflows/master_build_merge.yml:28]: That contains TessellDevelopment/REPo so to avoid the hassle to split then use

File: .github/workflows/master_build_merge.yml
Comment (varenyatessell) [reply .github/workflows/master_build_merge.yml:105]: That contains TessellDevelopment/Repo

File: .github/workflows/master_build_pr.yml
Comment (varenyatessell) [reply .github/workflows/master_build_pr.yml:81]: Okay

File: .github/workflows/master_pr.yml
Comment (varenyatessell) [reply .github/workflows/master_pr.yml:71]: https://github.com/TessellDevelopment/sample-event-driven-microservices/actions/runs/7195020228

File: .github/workflows/master_read_codeowners.yml
Comment (sundeeptessell) [inline .github/workflows/master_read_codeowners.yml:38]: Use `GITHUB_REPOSITORY_OWNER` env variable (available by default) instead of hardcoding `TessellDevelopment`; Update in all files wherever this comment holds good

File: .github/workflows/master_post_message.yml
Comment (sundeeptessell) [inline .github/workflows/master_post_message.yml:33]: Use default env variable  `GITHUB_REPOSITORY_OWNER`

File: .github/workflows/master_post_message.yml
Comment (sundeeptessell) [inline .github/workflows/master_post_message.yml:47]: Remove hardcoded `TessellDevelopment`

File: .github/workflows/master_post_message.yml
Comment (sundeeptessell) [inline .github/workflows/master_post_message.yml:42]: Use `GITHUB_API_URL` env variable for `https://api.github.com`

File: .github/workflows/master_check_approval.yml
Comment (sundeeptessell) [inline .github/workflows/master_check_approval.yml:32]: GITHUB_API_URL

File: .github/workflows/master_approve.yml
Comment (sundeeptessell) [inline .github/workflows/master_approve.yml:28]: Say, there is a team X with 2 or more members in it. And one member approves while another rejects the PR. 

What will be the overall status on the pending approval list or mergeability of the PR. 

Expected is that the PR should not be allowed to merge.

File: .github/workflows/master_check_approval.yml
Comment (sundeeptessell) [inline .github/workflows/master_check_approval.yml:51]: Remove hardcoding.

File: .github/workflows/master_check_approval.yml
Comment (sundeeptessell) [inline .github/workflows/master_check_approval.yml:59]: Do we need this in loop? It can be called outside loop, right

File: .github/workflows/master_check_approval.yml
Comment (sundeeptessell) [inline .github/workflows/master_check_approval.yml:66]: I see we are not considering the rejection. If one member rejects, then PR must not be allowed to merge.

File: .github/workflows/master_merge.yml
Comment (sundeeptessell) [inline .github/workflows/master_merge.yml:1]: In current implementation, our build workflow has `workflow_dispatch` along with `pull_request` event. This means build can be triggered manually if required. 

This (workflow_dispatch) gets used today, in the label creation process to trigger the build after tag creation so that new artifact gets created and pushed.  How are we planning to achieve this with master-ci framework.

However I don't see this option with master CI.

File: .github/workflows/master_build_merge.yml
Comment (varenyatessell) [inline .github/workflows/master_build_merge.yml:141]: Add Build command for redis-service

**Review by sundeeptessell — APPROVED** [review]

**Pull request merged**
